### PR TITLE
Add Anaconda Environment file

### PR DESCRIPTION
**Pull request by bcaddy**
# Summary

To avoid the issue with it being unclear which versions of which packages work with Iseult this adds an `environment.yml` file which indicates which versions are used and allows easy creation of an anaconda environment with the proper versions of the packages. I also updated the readme with new instructions for using this.

## Other

I also updated the `.gitignore` file to ignore a bunch of standard files and directories that shouldn't be tracked.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ IDL script.
 
 UPDATES:
 -------
+July 2023: Ported to Python 3.11, see environment.yml for details on all dependencies.
```
[the remark this answers is not in the document]
Comment (bcaddy) [reply README.md]: Fixed it, thanks

**Comment by bcaddy** [comment]
@ntoles, I added some more setup information and options. Better now?

File: README.md [hunk 2]
```diff
@@ -21,27 +21,36 @@ here and there, but I think it is working.
...
 
-To run Iseult on tigressdata or stellar vis nodes type the following:
+## Setup & Running
+
+There are two different methods to setup what you need for Iseult to run. The recommended method is to setup a python environment with conda, pyenv, etc and a conda `environment.yml` file is provided. The other method for those with access to the Stellar cluster at Princeton is to simply load the proper Anaconda module.
 ```bash
-# Initial setup, optional if your python environment has the correct version of all dependencies
+# Recommended setup
+# If on Stellar
 $ module load anaconda3/2024.6 # or the latest version
```
Comment (ntoles) [inline README.md:39]: The code does not work with the latest version, right? Or it does not matter if environment is used?

File: README.md [hunk 3]
```diff
@@ -21,27 +21,36 @@ here and there, but I think it is working.
...
+# If not on Stellar then install/load Anaconda
 $ cd /path/to/Iseult/
-$ chmod +x ./iseult.py
 $ conda env create
 $ conda activate iseult
```
Comment (ntoles) [inline README.md:43]: how does this know that environment.yml is ot be used for this environment named iseult? Is environment.yml a default file name that it looks for?

File: README.md [hunk 4]
```diff
@@ -21,27 +21,36 @@ here and there, but I think it is working.
...
+
+# Stellar simple setup
+$ module load anaconda3/2023.3
+$ cd /path/to/Iseult/
+$ chmod +x ./iseult.py
 
 # After initial setup
-$ module load anaconda3/2024.6 # or the latest version
-$ conda activate iseult
+# make sure the proper environment is activated or load the proper Anaconda
```
Comment (ntoles) [inline README.md:52]: how do I make sure that the proper environment is activated? please list the command I need to run

File: README.md (hunk 2 above)
Comment (bcaddy) [reply README.md:39]: It doesn't matter if an environment is used. 

Anaconda is a package manager (Conda) plus a bunch of packages commonly used for data science. Making a Conda environment uses Conda to create a new environment with whatever package versions you want.

File: README.md (hunk 3 above)
Comment (bcaddy) [reply README.md:43]: `environment.yml` is the default name that `conda env create` looks for. Specific files can be indicated with the `-f` flag.

File: README.md (hunk 4 above)
Comment (bcaddy) [reply README.md]: I added instructions

File: README.md [hunk 5]
```diff
@@ -10,30 +10,51 @@ Written by:
...
 $ cd /path/to/Iseult/
 $ chmod +x ./iseult.py
+
+# After initial setup
+# make sure the proper environment is activated
+$ conda activate iseult
+# or, if on Stellar you may load the proper Anaconda module
+$ module load anaconda3/2023.3
+# Then run iseult with
+$ cd /path/to/Iseult/
```
Comment (ntoles) [inline README.md:57]: It's minor, but I prefer recommending to 
cd to data directory
and
/path/to/Iseult/iseult.py &
this minimizes the number of clicks one has to do in GUI. People don't usually figure this out, and keep clicking.

Comment (ntoles) [reply README.md:57]: Note & -- people block their terminals without it...

File: README.md [hunk 6]
```diff
@@ -21,29 +21,45 @@ here and there, but I think it is working.
...
+
+# Stellar simple setup
+$ module load anaconda3/2023.3
+$ cd /path/to/Iseult/
+$ chmod +x ./iseult.py
 
 # After initial setup
-$ module load anaconda3/2024.6 # or the latest version
+# make sure the proper environment is activated
 $ conda activate iseult
```
Comment (ntoles) [inline README.md:53]: can this command be executed from any directory or do I need to cd to Iseult first?

Comment (bcaddy) [reply README.md:53]: Both the `module load` and the `conda activate` commands can be run from anywhere.

**Pull request merged**